lib/internal/errors.js

```diff
@@ -756,6 +756,7 @@ E('ERR_INVALID_CHAR',
     }
     return msg;
   }, TypeError);
+E('ERR_INVALID_COMMAND', '%s', TypeError);
```

Comment:
So… 
- I feel like this should definitely be a REPL-specific error code if it’s a REPL-specific condition – maybe `ERR_REPL_INVALID_COMMAND`?
- That being said, the condition in which this error is being emitted isn’t even about a command, it’s about a JS statement. How about `ERR_REPL_UNCAUGHT_EXCEPTION_HANDLER`? I don’t think we need to be afraid to get too specific with error codes.

Reply:
Hm, I feel most error codes are to specific and my goal here was to explicitly keep it more generic. We could reuse this error code in other places as well (while I do not have a strong opinion about the actual error code and e.g., `ERR_INVALID_STATEMENT` would also be fine for me).

The reason that I prefer less specific error codes is that in most cases users should not have to worry about adding lots of different error codes but to check only for a few which all go down to a similar error condition. That's why I also already thought about adding a kind of grouping property to our errors.

Reply:
What I mean is adding something similar like: https://github.com/petkaantonov/core-error-predicates

This module allows user code to catch specific types of errors in a more general fashion instead of checking for each error code specifically. It would e.g., be easier for users to just check for a `FileSystemError` instead of lots of different error codes that relate to file systems when they only want to make sure that a file is written. They might have a fallback mechanism which would just send the file somewhere else or report the error right away.
And even in such cases it would still be possible to check for individual error codes if they occur more frequently.

We also have a couple of errors which could e.g., all be categorized as "user input error". The user would still see what exact error happened but the user would only have to check for a single error type instead of lots of different specific ones.

I hope that makes my intention clearer.

Reply:
Fwiw, there is some existing discussion in https://github.com/nodejs/node/issues/8342 :)

Reply:
I switched the error code to `ERR_INVALID_REPL_INPUT`.